### PR TITLE
`StringToListConverter`: Update XML Documentation, Change `Separators` to Read-Write Property, Add Missing Unit Tests

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

This PR implements the following for `StringToListConverter`:
- `public string Separator`
  - Updates XML Documentation to explain the precedence of `ConverterParameter` and `Separators` 
  - Adds `null` validation
- `public List<string> Separators`
  - Change from Read-only Property, `{ get; }`, to  Read-Write Property, `{ get; set; }`.
  - Updates XML Documentation to explain the precedence of `ConverterParameter`
  - Adds `null` validation

### Additional information ###

The updated XML documentation is inspired by the MicrosoftDocs PR: https://github.com/MicrosoftDocs/CommunityToolkit/blob/73bd78afad79aa07c9c8329119ca45b5688d54d4/docs/maui/converters/string-to-list-converter.md

### CODE REVIEW

**Review by VladislavAntonyuk — APPROVED** [review]

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs [hunk 1]
```diff
@@ -8,15 +7,41 @@ namespace CommunityToolkit.Maui.Converters;
...
+	/// If ConverterParameter is null, this property will be used to delimit the substrings
 	/// </summary>
-	public IList<string> Separators { get; } = new List<string>();
+	public IList<string> Separators
+	{
+		get => separators;
+		set
+		{
+			ArgumentNullException.ThrowIfNull(value);
+
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:36]: Just curious if empty string is a valid separator

**Review by VladislavAntonyuk — APPROVED** [review]

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs (hunk 1 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:36]: Good point! I'll changes this verify against `string.IsNullOrEmpty` 👍

Comment (brminnick) [reply src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:49]: Done ✅

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs [hunk 2]
```diff
@@ -8,15 +7,61 @@ namespace CommunityToolkit.Maui.Converters;
...
+			foreach (var stringValue in value)
+			{
+				if (stringValue is null)
+				{
+					throw new ArgumentNullException(nameof(value), $"{nameof(value)} cannot contain null strings");
+				}
+				else if (string.IsNullOrEmpty(stringValue))
+				{
+					throw new ArgumentException("An empty string is not a vald separator", nameof(value));
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:58]:
```suggestion
					throw new ArgumentException("An empty string is not a valid separator", nameof(value));
```

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs [hunk 3]
```diff
@@ -40,15 +85,30 @@ public override IEnumerable<string> ConvertFrom(string? value, Type targetType,
...
 		{
+			foreach (var stringValue in separators)
+			{
+				if (stringValue is null)
+				{
+					throw new ArgumentNullException(nameof(value), $"{nameof(value)} cannot contain null strings");
+				}
+				else if (string.IsNullOrEmpty(stringValue))
+				{
+					throw new ArgumentException("An empty string is not a vald separator", nameof(value));
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:96]:
```suggestion
					throw new ArgumentException("An empty string is not a valid separator", nameof(value));
```

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs [hunk 4]
```diff
@@ -40,15 +85,30 @@ public override IEnumerable<string> ConvertFrom(string? value, Type targetType,
...
+
 			return Split(value, separators);
 		}
-
-		if (parameter is string separator)
+		else if (parameter is string separator)
 		{
+			if (string.IsNullOrEmpty(separator))
+			{
+				throw new ArgumentException("An empty string is not a vald separator", nameof(value));
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:106]:
```suggestion
				throw new ArgumentException("An empty string is not a valid separator", nameof(value));
```

**Review by VladislavAntonyuk — DISMISSED** [review]

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs [hunk 5]
```diff
@@ -8,15 +7,61 @@ namespace CommunityToolkit.Maui.Converters;
...
+		set
+		{
+			if (value is null)
+			{
+				throw new ArgumentNullException(nameof(value));
+			}
+
+			if (string.IsNullOrEmpty(value))
+			{
+				throw new ArgumentException("An empty string is not a vald separator", nameof(value));
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:31]:
```suggestion
				throw new ArgumentException("An empty string is not a valid separator", nameof(value));
```

**Review by VladislavAntonyuk — DISMISSED** [review]

**Review by VladislavAntonyuk — APPROVED** [review]

File: src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs [hunk 6]
```diff
@@ -8,15 +7,61 @@ namespace CommunityToolkit.Maui.Converters;
...
+				throw new ArgumentException("An empty string is not a valid separator", nameof(value));
+			}
+
+			separator = value;
+		}
+	}
 
 	/// <summary>
-	/// The strings that delimits the substrings in this string.
+	/// The strings that delimits the substrings in this string
```
Comment (bijington) [inline src/CommunityToolkit.Maui/Converters/StringToListConverter.shared.cs:39]: Minor but non plural delimit.

```suggestion
	/// The strings that delimit the substrings in this string
```

**Review by bijington — DISMISSED** [review]

**Review by VladislavAntonyuk — APPROVED** [review]

**Pull request merged**